demosplan/DemosPlanCoreBundle/Command/Data/DeleteProcedureCommand.php

```diff
@@ -0,0 +1,479 @@
+
+            // Procedure Behavior Definition
+            $this->output->writeln('Deleting Behavior Definitions');
+            $this->deleteBehaviorDefinitions();
+
+            // Procedure UI Definition
+            $this->output->writeln('Deleting UI Definitions');
+            $this->deleteUiDefinitions();
+
+            // form definitions -> field definitions
+            $this->output->writeln('Deleting Form Definitions');
+            $this->processFormDefinitions();
+
+            // delete gis layers
+            $this->output->writeln('Deleting Gis Layers');
+            $this->processGisLayers();
+
+            // delete procedure news
+            $this->output->writeln('Deleting News');
+            $this->deleteProcedureNews();
+
+            // delete tag topics -> tags
+            $this->output->writeln('Deleting Tags');
+            $this->processTags();
+
+            // delete predefined text categories -> predefined texts
+            $this->output->writeln('Deleting Predefined Texts');
+            $this->processPredefinedTexts();
+
+            // delete draft statements
+            $this->output->writeln('Deleting Draft Statements');
+            $this->deleteDraftStatements();
+
+            // delete import_emails -> attachments
+            $this->output->writeln('Deleting Import Emails');
+            $this->processImportEmails();
+
+            // delete hashed queries
+            $this->output->writeln('Deleting hashed Queries');
+            $this->deleteHashedQueries();
+
+            // delete remaining procedure files
+            $this->output->writeln('Deleting Procedure Files');
+            $this->deleteFromTableByIdentifierArray('_files', 'procedure_id', [$this->procedureId]);
+
+            // delete procedure report entries
+            $this->output->writeln('Deleting Report Entries');
+            $this->deleteReportEntriesByIdentifierAndType('procedure', [$this->procedureId]);
+
+            // delete procedure itself
+            $this->output->writeln('Deleting Procedure');
+            $this->deleteProcedure();
+
+            // reactivate foreign key checks
+            $this->output->writeln('Activate FK Checks');
+            $this->activateForeignKeyChecks();
+
+            $this->output->writeln("Procedure $this->procedureId was purged successfully!");
+
+            return Command::SUCCESS;
+        } catch (Exception $e) {
+            $this->output->error($e->getMessage());
+            $this->output->error($e->getTraceAsString());
+
+            return Command::FAILURE;
+        }
+    }
+
+    /**
+     * Find all statements + segments
+     * Iterate over all Statements and delete related stuff like meta, attachments, etc.
+     *
+     * @throws Exception
+     */
+    private function processAllStatements(): void
+    {
+        $statementIds = array_column($this->fetchFromTableByProcedure(['_st_id'], '_statement', '_p_id'), '_st_id');
+
+        print_r($statementIds);
```

Comment:
debug statement should be removed or $output used

Reply:
done in [86915ab](https://github.com/demos-europe/demosplan-core/pull/1958/commits/86915ab1e3d524cbe7f84dde906ed09fd0352f95)